src/libNetwork/P2PComm.cpp

```diff
@@ -68,7 +67,36 @@ static void close_socket(int* cli_sock)
     }
 }
 
-P2PComm::P2PComm() {}
+P2PComm::P2PComm()
+{
+    auto func = [this]() -> void {
+        while (true)
+        {
+            this_thread::sleep_for(chrono::seconds(BROADCAST_INTERVAL));
+            lock(m_broadcastToRemovedMutex, m_broadcastHashesMutex);
+            lock_guard<mutex> g(m_broadcastToRemovedMutex, adopt_lock);
+            lock_guard<mutex> g2(m_broadcastHashesMutex, adopt_lock);
+
+            for (unsigned int i = 0; i < BROADCAST_INTERVAL; ++i)
+            {
+                auto it = m_broadcastToRemoved.find(time(nullptr)
```

Comment:
I am a little concerned about the typecasting here. For eg. what would happen if BROADCAST_EXPIRTY + i > time(nullptr) ?